Sources/Administrator/InputHandler.cpp

```diff
@@ -101,22 +154,304 @@ void InputHandler::sendProcess(const string& full_command)
   }
 
   string process_name = full_command.substr(full_command.find_first_of(" ")+1);
+  if( process_name.find_first_of(" ") == string::npos)
+  {
+      cout << "Invalid number of arguments. The correct call is: send_process <name> <path to file>. \n";
+      cout << "Please try again. \n";
+      return;
+  }
+  string process_path = process_name.substr(process_name.find_first_of(" ")+1);
+  //checking if file path ends with .sh
```

Comment:
What if process_path is just one character? We're indexing string out of bounds.(because you're checking -1, -2,-3 indexes). 
And it doesn't have to be fille with .sh extension.